File: DI/DI-Dagger-Java/src/main/java/model/casa/Ventana.java
package model.casa;

import javax.inject.Inject;
import java.util.UUID;

public class Ventana {
    private final String id = UUID.randomUUID().toString();

    @Inject // Esta vez lo hago aquí porque no voy a usar un módulo
    public Ventana() {

    }

    public void abrir() {
        System.out.println("Abriendo ventana");
    }

    public void cerrar() {
        System.out.println("Cerrando ventana");
    }

    @Override
    public String toString() {
        return "Ventana{" +
                "id='" + id + '\'' +
                '}';
    }

}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/casas/Ventana.kt
package models.casas

import java.util.*
import javax.inject.Inject


class Ventana
@Inject constructor() {
    val id: String = UUID.randomUUID().toString()
    private var isOpened: Boolean = false
    fun abrir() {
        println("Abriendo ventana")
        isOpened = true
    }

    fun cerrar() {
        println("Cerrando ventana")
        isOpened = false
    }

    override fun toString(): String {
        return "Ventana(id='$id', isOpened=$isOpened)"
    }
}